enigma/src/main/java/module-info.java

```diff
@@ -0,0 +1,32 @@
+module cuchaz.enigma {
```

Comment:
Just declare this as `open module` to export everything, I suggest

Reply:
Ah, didnt know that was a thing 👍 Need to look at java modules in detail a little more at somepoint. IIRC idea did most of this.

Reply:
Hmm, open is actually opening for reflection after some lookup. We still need to export all packages explicitly (but opening stuff is still not bad imo)